### PR TITLE
Adds rootDir omission options

### PR DESCRIPTION
**Pull request by natedanner**
Adds rootDir option to allow omission of this setting from the name of the template.

### CODE REVIEW

File: tests/index-test.js [hunk 1]
```diff
@@ -54,3 +54,44 @@ test.serial.cb('preserveWhitespace', (t) => {
...
+    eval(compiled);
+    t.true(compiled.indexOf("(function(dust){dust.register(\"tests\\/_fixtures_\\/templateDir\\/rootDirTemplate\",body_0);") !== -1)
+    t.is(t.context.result, '<p>\n  Hello,\n  World!\n</p>\n')
+    t.end();
+  });
+});
+
+test.serial.cb('rootDirDefined', (t) => {
+  const fs = new MemoryFS();
+  WEBPACK_CONFIG.module.rules[0].options = {
```
Comment (avaly) [inline tests/index-test.js]: Can you please remove this options, since you're setting a new one just below?

File: tests/index-test.js [hunk 2]
```diff
@@ -54,3 +54,44 @@ test.serial.cb('preserveWhitespace', (t) => {
...
+    rootDir: 'tests/_fixtures_/templateDir'
+  };
+  const compiler = webpack(WEBPACK_CONFIG);
+  compiler.outputFileSystem = fs;
+  compiler.run(() => {
+    const compiled = fs.readFileSync('/compiled.js', 'utf-8');
+    eval(compiled);
+    t.true(typeof compiled === 'string')
+    t.true(compiled.indexOf("(function(dust){dust.register(\"rootDirTemplate\",body_0);") !== -1)
+    t.true(compiled.indexOf("(function(dust){dust.register(\"tests\/_fixtures_\/templateDir\/rootDirTemplate\",body_0);") === -1)
```
Comment (avaly) [inline tests/index-test.js]: There seems to be a mismatch between this assertion and the similar one in the previous test: the first one is using `\\/` and this one is using `\/` only. Please use the same escaping sequence.

File: README.md [hunk 3]
```diff
@@ -41,7 +41,23 @@ You can also preserve the whitespace in your template with the following loader
 {
   test: /\.dust$/,
   loader: 'dust-loader',
-  query: 'preserveWhitespace'
+  options: 'preserveWhitespace'
```
Comment (avaly) [inline README.md]: Can you please also add an example of using `preserveWhitespace` unsing `options` as an object?

**Comment by natedanner** [comment]
Changes requested made. Cheers!

File: README.md (hunk 3 above)
Comment (natedanner) [inline README.md]: Example added.

File: tests/index-test.js (hunk 1 above)
Comment (natedanner) [inline tests/index-test.js]: Removed duplicate code.

File: tests/index-test.js (hunk 2 above)
Comment (natedanner) [inline tests/index-test.js]: Fixed escaping.

**Pull request merged**